### ISSUE TITLE
Underdark feedback for developers of React bindings

### ISSUE THREAD

**Issue by virl** [alexkendall/react-native-bluetooth-cross-platform#8]
Greetings!
My nickname is Virl and I'm author of Underdark libraries.

Feel free to ask me here any questions about project or give feedback and proposals regarding React interoperability from native side. You can also do it in Issues of Underdark repositories on GitHub, if you feel it will be more appropriate that way.

Wish all the best for your bindings project!

**Comment by butchmarshall** [comment]
Hi Virl  thanks for releasing this!

Given the projects current state would it be possible to send voice data over the connection?  I'm interested in using this to make a local walkie talkie type application.

**Comment by virl** [comment]
It should be possible, but underlying mobile OS uses tcp-like connection, so I don’t know how good latency would be.

**Comment by butchmarshall** [comment]
That's great @virl!

I'm trying to use [react-native-webrtc](https://github.com/oney/react-native-webrtc) for the connections but don't even know where to get started with this library to allow the devices to discover one another?

Do you have any pointers to get me started?

**Comment by pttd1304** [comment]
Hi @virl !
Can you give us some instructions on how to use the library? Like where I should start or something.
A sample project would be great.

**Comment by virl** [comment]
@pttd1304 Hi. Library's github repo contains sample project that transfers arbitrary bytes.

**Comment by pttd1304** [comment]
@virl Hey. Can you provide a link to the sample project, please? I cannot find it anywhere. Many thanks.

**Comment by virl** [comment]
@pttd1304 
https://github.com/udark/underdark-cocoa/
https://github.com/udark/underdark-android/

**Comment by pttd1304** [comment]
@virl Thank you. I thought it was a sample project for react native underdark. How should I start a project with react native underdark library?

**Comment by virl** [comment]
@pttd1304 Sorry, React native support for it is not my project.

**Comment by pttd1304** [comment]
@virl Oh. So may I ask what this react native bluetooth crossplatform is for, and how I can utilize this library.

**Comment by virl** [comment]
@pttd1304 It is not my project, so I guess you should not ask that from me. :)

**Comment by pttd1304** [comment]
@virl Ok. Thanks for the support. Time to get my hands dirty then.

**Comment by MichaelFBA** [comment]
@virl 
Query about WIFI connections, do users need to connect / pair to each other to pass information between the mesh network?

**Comment by MichaelFBA** [comment]
@virl 
Just a couple of quick questions re: WIFI connections
1. Is there a limit to the number of users connected?
2. Once I am connected to a number of users are we using the same IP range on the adhoc network?

**Comment by virl** [comment]
@MichaelFBA 
1. For Wi-Fi connections, its OS socket fd limit (thousands in same network).
2. Adhoc is not supported on stock Android and my library doesn't use it. For Wi-Fi it connect only in same Wi-Fi LAN Access Point.

**Comment by michaeldanielbell** [comment]
@virl 
Thanks for clearing that up. I was under the impression we could use your library without an intermediary access point as I think that's where the real magic happens.

I noticed Android has Wifi Direct, is this something you could use?
https://developer.android.com/guide/topics/connectivity/wifip2p.html

**Comment by virl** [comment]
@michaeldanielbell No, Wi-Fi Direct on Android doesn't work because it is bugged beyond repair.
